llvm/include/llvm/CAS/CASID.h

```diff
@@ -71,65 +67,52 @@ class CASID {
   }
   std::string toString() const;
 
-  ArrayRef<uint8_t> getHash() const { return getContext().getHashImpl(*this); }
-
-  friend bool operator==(CASID LHS, CASID RHS) {
-    // If it's the same CAS (or both nullptr), then the IDs are directly
-    // comparable.
-    if (LHS.Context == RHS.Context)
-      return LHS.InternalID == RHS.InternalID;
+  ArrayRef<uint8_t> getHash() const;
 
-    // Check if one CAS is nullptr, indicating a tombstone or empty key for
-    // DenseMap, and return false if so.
+  friend bool operator==(const CASID &LHS, const CASID &RHS) {
+    // EmptyKey or TombstoneKey.
     if (!LHS.Context || !RHS.Context)
       return false;
```

Comment:
Oh I get it, in the old code it first checked for `if (LHS.Context == RHS.Context) return LHS.InternalID == RHS.InternalID;`.  I think we still need that to handle empty/tombstone keys.  Now it could be ``if (LHS.Context == RHS.Context) return LHS.Hash == RHS.Hash;` since we don't need to check the hash scheme if it's the same context.